packages/promise-kit/src/promiseKit.js

```diff
@@ -7,14 +7,25 @@
 /** @type {import('@agoric/eventual-send').HandledPromiseConstructor | PromiseConstructor} */
 const BestPipelinablePromise = globalThis.HandledPromise || Promise;
 
+// Using freeze if harden is not available allows this module to be usable both
+// with and without hardened JavaScript.
+const harden = globalThis.harden || /** @type {typeof globalThis.harden} */(Object.freeze);
+
 /**
  * @template T
- * @typedef {Object} PromiseRecord A reified Promise
+ * @typedef {Object} PromiseKit A reified Promise
  * @property {(value: ERef<T>) => void} resolve
  * @property {(reason: any) => void} reject
  * @property {Promise<T>} promise
  */
 
+/**
+ * PromiseRecord is deprecated in favor of PromiseKit.
+ *
+ * @template T
+ * @typedef {PromiseKit<T>} PromsieRecord
```

Comment:
```suggestion
 * @typedef {PromiseKit<T>} PromiseRecord
```

And doesn't typescript allow explicit deprecation?  That's just _dicta_

Reply:
It does. I’ll check how well it interacts with tooling.

I’ve found in the past that tooling that sees _and enforces_ deprecation somewhat runs afoul of the intent of providing a path for gradual migration. I’ll check.